### ISSUE TITLE
extra error on recur with cond->

### ISSUE THREAD

**Issue by ikitommi** [clj-kondo/clj-kondo#1545]
**version**

v2022.01.13

**platform**

JVM

**problem**

extra error on `recur` with `cond->`

**repro**

```bash
➜  ~ echo '(loop [x 10] (cond-> (dec x) (pos? x) (recur)))' | clj
Clojure 1.10.3
user=> -1
user=>
➜  ~ echo '(loop [x 10] (cond-> (dec x) (pos? x) (recur)))' | clj-kondo --lint -
<stdin>:1:39: error: Recur can only be used in tail position.
linting took 9ms, errors: 1, warnings: 0
```

**expected behavior**

No error
